src/main/java/com/google/sps/data/UserQuery.java

```diff
@@ -59,11 +59,22 @@ public UserQuery(String query, String tags, String authors) {
     ArrayList<Query.Filter> searchFilters = new ArrayList<Query.Filter>();
     
     // Creates and adds filters to the list if there is data to make the filter
-    if (tagsAndIngredients.size() > 0){
+    if (tagsAndIngredients.size() > 1){
       searchFilters.add(createTagsFilter());
+    } else if (tagsAndIngredients.size() == 1){
+      // Use of collection interface methods only to allow for future flexibility
+      String tempTag = tagsAndIngredients.iterator().next();
+      searchFilters.add(createSingleTagFilter(tempTag));
     }
-    if (authors.size() > 0) {
+    if (authors.size() > 1) {
       searchFilters.add(createAuthorsFilter());
+    } else if (authors.size() == 1) {
+      // Use of collection interface methods only to allow for future flexibility
+      String tempAuthor = authors.iterator().next();
```

Comment:
nit: can you change all the variable names with "temp" to more meaningful variable names? Like this could be `firstAuthor` or `singleAuthor`, for example

Reply:
Ack and changed